apps/opik-backend/src/main/java/com/comet/opik/domain/SpanDAO.java

```diff
@@ -444,6 +493,102 @@ public Mono<Void> insert(@NonNull Span span) {
                 .then();
     }
 
+    @Trace(dispatcher = true)
+    public Mono<Void> batchInsert(@NonNull List<Span> spans) {
+        return Mono.from(connectionFactory.create())
+                .flatMapMany(connection -> {
+                    if (spans.isEmpty()) {
+                        return Mono.just(0L);
+                    }
+
+                    return insert(spans, connection);
+                })
+                .then();
+    }
+
+    private Publisher<? extends Result> insert(Collection<Span> spans, Connection connection) {
```

Comment:
It's usually better to use more specific collections. Honestly, probably the best approach here is to use a Set of spans all the way from the request object to every method in the path (service methods, DAO methods etc.).

Reply:
I used collection here because we don't need any particular feature from the List of Sets, so I went generic to make it reusable. But we can change it, no worries. However, regarding using Set on APIs, I have a few concerns. First, if the client sends a list with duplicates, the set will use the first and discard the rest. I believe it would be more reasonable to return 400 in this case. The other point would be what to use in the equals and hash code (Only id, all the sortable keys, the whole object).